akka-typed/src/main/scala/akka/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -75,16 +75,20 @@ import akka.util.OptionVal
       a.SupervisorStrategy.Stop
   }
 
-  override def preStart(): Unit =
+  override def preStart(): Unit = {
     behavior = validateAsInitial(undefer(behavior, ctx))
+    if (!isAlive(behavior)) context.stop(self)
```

Comment:
the situation should be improved somewhat when ticket https://github.com/akka/akka/issues/22841 is done